### PR TITLE
Add okta auth 14

### LINKED ISSUE #388: Add support for Okta auth
Corp would like to add Okta auth to Grafana, Ill add it and merge it back upstream.

Add section from https://grafana.com/docs/grafana/latest/auth/okta/#enable-okta-oauth-in-grafana to mimic other auth providers

### PR DESCRIPTION
**Pull request by mbaitelman**
# Description

Add support for Okta auth

## Issues Resolved
#388 

## Check List

- [ ] All tests pass. See TESTING.md for details.
- [ ] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by mbaitelman** [comment]
Im still in middle of updating this for work but figured Id get an early set of eyes on it.

**Comment by xorima** [comment]
looks at a high level like it is going down the right path to me. The build errors will need to be fixed though :)

File: resources/config_auth_okta.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#
+# Configures Okta auth for installed grafana instance
+
+property  :instance_name,        String,         name_property: true
+property  :auth_name,            String,         default: 'Okta'
+property  :enabled,              [true, false],  default: false
+property  :allow_sign_up,        [true, false],  default: true
+property  :client_id,            String,         default: ''
+property  :client_secret,        String,         default: ''
+property  :scopes,               String,         default: '', description: 'openid profile email groups'
```
Comment (xorima) [inline resources/config_auth_okta.rb]: If this is a list to be allowed either keys or validation could be good here

**Comment by xorima** [comment]
@mbaitelman  is the Okta auth broken in it's current implementation?

**Comment by mbaitelman** [comment]
Before these changes? It didnt exist.
The code here works internally (though we are dealing with other unrelated issues first, which is why I haven't finished this PR)

**Comment by bmhughes** [comment]
Implemented by #403

**Pull request closed without merging**